app/types/types.ts

```diff
@@ -0,0 +1,167 @@
+import { ReactNode } from "react";
+import { Variables } from "graphql-request";
+import { boolean } from "zod";
+
+export interface LookupTableDef {
+  tableSchema: "public" | "lookups";
+  tableName: string;
+}
+
+export interface LookupTableOption {
+  id: number;
+  label: string;
+}
+
+export type InputType = "text" | "number" | "yes_no" | "select";
+
+/**
+ * The base definition of a database column referenced by our app — where
+ * <T> should be one of our core record types (Crash, Person, etc)
+ */
+export type ColBaseDef<T extends Record<string, unknown>> = {
```

Comment:
`ColBaseDef` is the smallest building of our types coming out of the database, and we can use them to build up any time of record: crashes, units, etc.

Also, you can see that I have extended from `Record<string, unknown>>` type, and i do that in many places. I think that's fine and good, but because it is used so often across the codebase we should probably give it a name and export it. something like `<RecordWithStringKeys>`. that's all it is: an object with keys that are always strings.

this will make more sense after reading about the `Record` utility type: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type  :)